rct229/rules/section23/section23rule8.py

```diff
@@ -0,0 +1,139 @@
+            super(Section23Rule8.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section23Rule8.HVACRule.SupplyFanRule(),
+                index_rmr="baseline",
+                list_path="$..supply_fans[*]",
+            )
+
+        class SupplyFanRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section23Rule8.HVACRule.SupplyFanRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={
+                        "$": [
+                            "design_airflow",
+                            "design_electric_power",
+                            "output_validation_points",
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                supply_fan_b = context.baseline
+
+                design_airflow_b = supply_fan_b["design_airflow"]
+                design_electric_power_b = supply_fan_b["design_electric_power"]
+                output_validation_points_b = supply_fan_b["output_validation_points"]
+
+                return {
+                    "design_airflow_b": CalcQ("air_flow_rate", design_airflow_b),
+                    "design_electric_power_b": CalcQ(
+                        "electric_power", design_electric_power_b
+                    ),
+                    "output_validation_points_b": output_validation_points_b,
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                design_airflow_b = calc_vals["design_airflow_b"]
+                design_electric_power_b = calc_vals["design_electric_power_b"]
+                output_validation_points_b = calc_vals["output_validation_points_b"]
+
+                output_validation_points = [
+                    [output["airflow"], output["result"]]
+                    for output in output_validation_points_b
+                ]
+
+                target_validation_points = [
+                    [
+                        SUPPLY_AIRFLOW_COEFF[idx] * design_airflow_b,
+                        DESIGN_POWER_COEFF[idx] * design_electric_power_b,
+                    ]
+                    for idx in range(11)
+                ]
+                place = 1
+                return all(
+                    list(
+                        map(
+                            lambda x, y: std_equal(x[0], y[0]),
+                            output_validation_points,
+                            target_validation_points,
+                        )
+                    )
+                    and list(
+                        map(
+                            lambda x, y: std_equal(x[1], y[1]),
+                            output_validation_points,
+                            target_validation_points,
+                        )
+                    )
+                )
```

Comment:
I like the solution written here - it is a good use of map function.
To further simply it, we could use zip function as well:
```python
all([std_equal(x[0], y[0] and std_equal(x[1], y[1]) for x, y in zip(output_validation_points, target_validation_points)
```
One issue is the length matching. Using map, we will need to ensure two have the same length, otherwise, it will crash.
Using zip, the unmatched portion will be silently skipped.

So we will need to add:
```python
len(output_validation_points) == len(target_validation_points)
```

Reply:
Great point! Thanks for letting me know about this!